### ISSUE TITLE
I get these [com.arjuna.ats.arjuna] warnings and right after the readiness probe dies

### ISSUE THREAD

**Issue by RumRogerz** [keycloak/keycloak#13656]
### Describe the bug

Good day, devs!

To be honest, I'm unsure if this is a bug or some sort of configuration that I made horribly wrong.

Setup:
Keycloak - kubernetes deployment
Infinispan - kubernetes deployment -  we use a seperate infinispan deployment as a remote-store for cache.
DB - cockroach

Keycloak (Quarkus) is configured with remote-stores on all distributed and replicated caches to a seperate infinispan deployment located on the same k8s cluster.  For the most part, this works!  We have started testing its usability and randomly, we will find that a user (me, mostly) will attempt a log into keycloak. Browser will spin for a bit and either give me a blank screen, endlessly trying to load the page or a 503.  When this happens, a random pod will have its readiness probe fail.  The readiness probe fails with an HTTP status code of 500

The logs that gather at this time show these warnings (shaved down for brevity):

```
2022-08-08 19:23:49,068 DEBUG [org.infinispan.CLUSTER] (jgroups-25,keycloak-2-196) ISPN000336: Finished rebalance for cache actionTokens, topology CacheTopology{id=97, phase=NO_REBALANCE, rebalanceId=28, currentCH=DefaultConsistentHash{ns=256, owners = (3)[keycloak-2-196: 84+86, keycloak-1-2895: 85+89, keycloak-0-14082: 87+81]}, pendingCH=null, unionCH=null, actualMembers=[keycloak-2-196, keycloak-1-2895, keycloak-0-14082], persistentUUIDs=[20aa54eb-c02d-4d19-b367-ef0ddceb6b80, 88d23310-ff4d-4d8d-94f3-bf7e7f68077a, 8acfeded-94e0-4dbc-8dba-36bd7307e668]}
2022-08-08 19:23:49,068 INFO  [org.infinispan.CLUSTER] (jgroups-25,keycloak-2-196) [Context=actionTokens] ISPN100010: Finished rebalance with members [keycloak-2-196, keycloak-1-2895, keycloak-0-14082], topology id 97
2022-08-08 19:27:51,264 WARN  [com.arjuna.ats.arjuna] (Transaction Reaper) ARJUNA012117: TransactionReaper::check processing TX 0:ffff0a2e05af:a433:62f16208:69 in state  RUN
2022-08-08 19:27:51,265 INFO  [com.arjuna.ats.arjuna] (Transaction Reaper Worker 0) ARJUNA012404: Action id 0:ffff0a2e05af:a433:62f16208:69 - thread Timer-0 at time 19:25:51.266 had stackTrace org.keycloak.cluster.infinispan.InfinispanNotificationsManager.lambda$notify$0(InfinispanNotificationsManager.java:168)
org.keycloak.cluster.infinispan.InfinispanNotificationsManager$$Lambda$1498/0x0000000840d10440.run(Unknown Source)
org.keycloak.common.util.Retry.executeWithBackoff(Retry.java:87)
org.keycloak.common.util.Retry.executeWithBackoff(Retry.java:79)
org.keycloak.cluster.infinispan.InfinispanNotificationsManager.notify(InfinispanNotificationsManager.java:159)
org.keycloak.cluster.infinispan.InfinispanClusterProvider.notify(InfinispanClusterProvider.java:130)
org.keycloak.models.sessions.infinispan.changes.sessions.CrossDCLastSessionRefreshStore.sendMessage(CrossDCLastSessionRefreshStore.java:55)
org.keycloak.models.sessions.infinispan.changes.sessions.AbstractLastSessionRefreshStore.checkSendingMessage(AbstractLastSessionRefreshStore.java:61)
org.keycloak.models.sessions.infinispan.changes.sessions.AbstractLastSessionRefreshStoreFactory.lambda$setupPeriodicTimer$0(AbstractLastSessionRefreshStoreFactory.java:45)
org.keycloak.models.sessions.infinispan.changes.sessions.AbstractLastSessionRefreshStoreFactory$$Lambda$1364/0x0000000840c3d040.run(Unknown Source)
org.keycloak.services.scheduled.ScheduledTaskRunner.runTask(ScheduledTaskRunner.java:73)
org.keycloak.services.scheduled.ScheduledTaskRunner.run(ScheduledTaskRunner.java:54)
org.keycloak.timer.basic.BasicTimerProvider$1.run(BasicTimerProvider.java:53)
java.base@11.0.16/java.util.TimerThread.mainLoop(Timer.java:556)
java.base@11.0.16/java.util.TimerThread.run(Timer.java:506)

[ The above INFO repeats about 3 more times, did not include because they are verbatim the same and a wall of text is never fun to sift through ]

2022-08-08 19:27:51,267 WARN  [com.arjuna.ats.arjuna] (Transaction Reaper Worker 0) ARJUNA012108: CheckedAction::check - atomic action 0:ffff0a2e05af:a433:62f16208:69 aborting with 1 threads active!
2022-08-08 19:27:51,267 WARN  [com.arjuna.ats.arjuna] (Transaction Reaper Worker 0) ARJUNA012121: TransactionReaper::doCancellations worker Thread[Transaction Reaper Worker 0,5,main] successfully canceled TX 0:ffff0a2e05af:a433:62f16208:69
````
I'm no Java guru.  But From what I am reading, something about a timer and something about infinispan, possibly the sessions cache (although I could be wrong) is not having a good time.  Infinispan logs show literally *nothing* of relevance. 

Loglevel is set as:
`error,org.keycloak.events:debug,org.infinispan.CLUSTER:debug,org.infinispan.HOTROD:debug,com.arjuna.ats.arjuna:debug`

### Version

19.0.1

### Expected behavior

I would imagine since I am not seeing errors and just warnings, that maybe this doesn't kill the server or service that is responsible for keeping things running?  I really wish I knew what was happening behind the scenes but I'm super lost, please excuse my very apparent ignorance.

### How to Reproduce?

Start a keycloak deployment and a seperate infinispan deployment:

Keycloak:
cache-ispn.xml
replace remote-server-host with your infinispan headless service fqdm.

```xml
<?xml version="1.0" encoding="UTF-8"?>

<infinispan
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="urn:infinispan:config:13.0 http://www.infinispan.org/schemas/infinispan-config-11.0.xsd"
        xmlns="urn:infinispan:config:13.0">

    <cache-container name="keycloak">
        <transport lock-timeout="60000"/>
        <local-cache name="realms">
            <encoding>
                <key media-type="application/x-java-object"/>
                <value media-type="application/x-java-object"/>
            </encoding>
            <memory max-count="10000"/>
        </local-cache>
        <local-cache name="users">
            <encoding>
                <key media-type="application/x-java-object"/>
                <value media-type="application/x-java-object"/>
            </encoding>
            <memory max-count="10000"/>
        </local-cache>
        <distributed-cache name="sessions" owners="2">
            <expiration lifespan="-1"/>
            <remote-store xmlns="urn:infinispan:config:store:remote:13.0"
                            cache="sessions" 
                            fetch-state="false"
                            purge="false"
            	            preload="false"
                            segmented="false"
            	            shared="true">
                <remote-server host="infinispan.infinispan-ap-southeast-1.svc.cluster.local" port="11222" />
                <property name="rawValues">true</property>
                <property name="marshaller">org.keycloak.cluster.infinispan.KeycloakHotRodMarshallerFactory</property>
            </remote-store></distributed-cache>
        <distributed-cache name="authenticationSessions" owners="2">
            <expiration lifespan="-1"/>
            <remote-store xmlns="urn:infinispan:config:store:remote:13.0"
                            cache="authenticationSessions" 
                            fetch-state="false"
                            purge="false"
            	            preload="false"
                            segmented="false"
            	            shared="true">
                <remote-server host="infinispan.infinispan-ap-southeast-1.svc.cluster.local" port="11222" />
                <property name="rawValues">true</property>
                <property name="marshaller">org.keycloak.cluster.infinispan.KeycloakHotRodMarshallerFactory</property>
            </remote-store></distributed-cache>
        <distributed-cache name="offlineSessions" owners="2">
            <expiration lifespan="-1"/>
            <remote-store xmlns="urn:infinispan:config:store:remote:13.0"
                            cache="offlineSessions" 
                            fetch-state="false"
                            purge="false"
            	            preload="false"
                            segmented="false"
            	            shared="true">
                <remote-server host="infinispan.infinispan-ap-southeast-1.svc.cluster.local" port="11222" />
                <property name="rawValues">true</property>
                <property name="marshaller">org.keycloak.cluster.infinispan.KeycloakHotRodMarshallerFactory</property>
            </remote-store></distributed-cache>
        <distributed-cache name="clientSessions" owners="2">
            <expiration lifespan="-1"/>
            <remote-store xmlns="urn:infinispan:config:store:remote:13.0"
                            cache="clientSessions" 
                            fetch-state="false"
                            purge="false"
            	            preload="false"
                            segmented="false"
            	            shared="true">
                <remote-server host="infinispan.infinispan-ap-southeast-1.svc.cluster.local" port="11222" />
                <property name="rawValues">true</property>
                <property name="marshaller">org.keycloak.cluster.infinispan.KeycloakHotRodMarshallerFactory</property>
            </remote-store></distributed-cache>
        <distributed-cache name="offlineClientSessions" owners="2">
            <expiration lifespan="-1"/>
            <remote-store xmlns="urn:infinispan:config:store:remote:13.0"
                            cache="offlineClientSessions" 
                            fetch-state="false"
                            purge="false"
            	            preload="false"
                            segmented="false"
            	            shared="true">
                <remote-server host="infinispan.infinispan-ap-southeast-1.svc.cluster.local" port="11222" />
                <property name="rawValues">true</property>
                <property name="marshaller">org.keycloak.cluster.infinispan.KeycloakHotRodMarshallerFactory</property>
            </remote-store></distributed-cache>
        <distributed-cache name="loginFailures" owners="2">
            <expiration lifespan="-1"/>
            <remote-store xmlns="urn:infinispan:config:store:remote:13.0"
                            cache="loginFailures" 
                            fetch-state="false"
                            purge="false"
            	            preload="false"
                            segmented="false"
            	            shared="true">
                <remote-server host="infinispan.infinispan-ap-southeast-1.svc.cluster.local" port="11222" />
                <property name="rawValues">true</property>
                <property name="marshaller">org.keycloak.cluster.infinispan.KeycloakHotRodMarshallerFactory</property>
            </remote-store></distributed-cache>
        <local-cache name="authorization">
            <encoding>
                <key media-type="application/x-java-object"/>
                <value media-type="application/x-java-object"/>
            </encoding>
            <memory max-count="10000"/>
        </local-cache>
        <replicated-cache name="work">
            <expiration lifespan="-1"/>
            <remote-store xmlns="urn:infinispan:config:store:remote:13.0"
                            cache="work" 
                            fetch-state="false"
                            purge="false"
            	            preload="false"
                            segmented="false"
            	            shared="true">
                <remote-server host="infinispan.infinispan-ap-southeast-1.svc.cluster.local" port="11222" />
                <property name="rawValues">true</property>
                <property name="marshaller">org.keycloak.cluster.infinispan.KeycloakHotRodMarshallerFactory</property>
            </remote-store></replicated-cache>
        <local-cache name="keys">
            <encoding>
                <key media-type="application/x-java-object"/>
                <value media-type="application/x-java-object"/>
            </encoding>
            <expiration max-idle="3600000"/>
            <memory max-count="1000"/>
        </local-cache>
        <distributed-cache name="actionTokens" owners="2">
            <encoding>
                <key media-type="application/x-java-object"/>
                <value media-type="application/x-java-object"/>
            </encoding>
            <expiration max-idle="-1" lifespan="-1" interval="300000"/>
            <memory max-count="-1"/>
            <remote-store xmlns="urn:infinispan:config:store:remote:13.0"
                            cache="actionTokens" 
                            fetch-state="false"
                            purge="false"
            	            preload="false"
                            segmented="false"
            	            shared="true">
                <remote-server host="infinispan.infinispan-ap-southeast-1.svc.cluster.local" port="11222" />
                <property name="rawValues">true</property>
                <property name="marshaller">org.keycloak.cluster.infinispan.KeycloakHotRodMarshallerFactory</property>
            </remote-store></distributed-cache>
    </cache-container>
</infinispan>
```
```yaml
keycloak:
  log_level: "error,org.keycloak.events:debug,org.infinispan.CLUSTER:debug,org.infinispan.HOTROD:debug,com.arjuna.ats.arjuna:debug"
  cache_type: "ispn"
  cache_stack: "kubernetes"
```

Infinispan:

```xml
<cache-container name="keycloak">
    <transport
        cluster="koolbeans_1"
        node-name="my-cool-node-name"
        stack="default" 
        site="ap-southeast-1"/>
<replicated-cache-configuration name="replicated" >
      <locking acquire-timeout="0" /> 
</replicated-cache-configuration>
    <replicated-cache name="work" configuration="replicated"/>
    <replicated-cache name="sessions" configuration="replicated"/>
    <replicated-cache name="authenticationSessions" configuration="replicated"/>
    <replicated-cache name="clientSessions" configuration="replicated"/>
    <replicated-cache name="offlineSessions" configuration="replicated"/>
    <replicated-cache name="offlineClientSessions" configuration="replicated"/>
    <replicated-cache name="actionTokens" configuration="replicated"/>
    <replicated-cache name="loginFailures" configuration="replicated"/>
</cache-container>
```

Open a bunch of webpages and log in and out of keycloak until it gets angry and 503's on you - or just endlessly tries to refresh resulting in a blank page.

**Comment by kkcmadhu** [comment]
i too face similar problems. my first time startup on keycloak took way long, after i increased the transaction timeout i was finally able to start keycloak 18.. but even after that i occassionly see my logs info/warning/errors related to these.. e.g 
https://keycloak.discourse.group/t/error-failed-to-run-scheduled-task-clearexpiredusersessions/16879

**Comment by martin-kanis** [comment]
@pedroigor Can this be related/ the same issue as this https://github.com/keycloak/keycloak/issues/14011?

**Comment by pedroigor** [comment]
@martin-kanis Yeah, it should be related.

@RumRogerz Could you please try the latest versions? It was included on 19.0.2+.
